haystack/components/preprocessors/csv_document_splitter.py

```diff
@@ -242,3 +261,16 @@ def _recursive_split(
                 result.append(table)
 
         return result
+
+    def _split_by_row(self, df: "pd.DataFrame") -> List["pd.DataFrame"]:
+        """Split each CSV row into a separate subtable"""
+        try:
+            split_dfs = []
+            for idx, row in enumerate(df.itertuples(index=False)):
+                split_df = pd.DataFrame(row).T
+                split_df.index = [idx]  # Set the index of the new DataFrame to idx
+                split_dfs.append(split_df)
+            return split_dfs
+        except Exception as e:
+            logger.warning("Error while splitting CSV rows to documents: {error}", error=e)
+            return []
```

Comment:
If there is a chance that `split_dfs` can be an empty list let's add an explicit check for it right before this line further up
```python
            # Sort split_dfs first by row index, then by column index
            split_dfs.sort(key=lambda dataframe: (dataframe.index[0], dataframe.columns[0]))
```
such as 
```python
if len(split_dfs) == 0:
    logger.warning("No sub-tables found while splitting CSV Document with id {doc_id}. Skipping document.", doc_id=document.id)
    continue
```